### PR TITLE
#161877976 Add update an article feature

### PR DESCRIPTION
**Pull request by TheoOkafor**
#### What does this PR do?
- Add Update an article feature

#### Description of Task to be completed?
- Create the route for updating an article
- Write an integration test for route
- Write documentation

#### How should this be manually tested?
- git clone `https://github.com/andela/haven-ah-backend.git`.
- Install `node` if not installed.
- ensure that PostgreSQL is running on your local machine.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.

To test on Postman
- run `npm run start-dev`
- Add the title, body and description as `x-www-form-urlencoded`
- On postman run PATCH `localhost:5000/api/articles/:slug/` (set JWT token as`x-access-token` in header).
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#161877976

#### Screenshots (if appropriate)

Good Response:
screen shot 2018-11-18 at 6 21 49 am

### CODE REVIEW

**Comment by ucheg6** [comment]
Good work Theo. This looks great.

File: server/controllers/article.js [hunk 1]
```diff
@@ -166,6 +166,35 @@ class Article {
...
+  static async updateArticle(request, response) {
+    const { id } = request.article;
+    const {
+      title,
+      body,
+      description,
+      images,
+    } = request.body;
+    const readtime = Math.floor(body.split(' ').length / 3);
+
```
Comment (PriscillaSam) [inline server/controllers/article.js:209]: @TheoOkafor are we updating every single field when the update function is called even when the only thing that changed is maybe the images or the title?

File: server/routes/article.js [hunk 2]
```diff
@@ -77,4 +77,12 @@ router.delete(
   tryCatchWrapper(Article.deleteArticle)
 );
 
+router.put(
+  '/articles/:slug',
+  isAuthenticated,
+  inputValidator.articleValidator,
+  checkIfArticleExists,
+  isAuthorized.article,
```
Comment (victor-nweze-deel) [inline server/routes/article.js]: This could be isAuthorized.checkAuthor

File: server/controllers/article.js (hunk 1 above)
Comment (TheoOkafor) [reply server/controllers/article.js:209]: Yes.

Comment (victor-nweze-deel) [reply server/controllers/article.js:209]: This is fine

Comment (seyi-adeleke) [reply server/controllers/article.js:209]: Technically,  what we're trying to do here is a `patch`. Please read on the difference between a `put` and a `patch`. Also you should be updating the updatedAt column as well

File: server/middlewares/authorization.js [hunk 3]
```diff
@@ -42,6 +42,24 @@ class authorization {
...
+
+  /**
+   * This checks whether the requester (user) is authorized
+   * to complete the request on an article.
+   * @param {object} request
+   * @param {object} response
+   * @param {function} next
+   * @returns {function} callback function
+   */
+  static async article(request, response, next) {
```
Comment (seyi-adeleke) [inline server/middlewares/authorization.js]: article is definetely not a good name for this middleware

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

File: server/controllers/article.js (hunk 1 above)
Comment (PriscillaSam) [reply server/controllers/article.js:209]: @seyi-adeleke the `updatedAt` column  is updated automatically by sequelize once an update is performed.

File: server/middlewares/authorization.js (hunk 3 above)
Comment (TheoOkafor) [reply server/middlewares/authorization.js]: You are right @seyi-adeleke. `patch` will be more appropriate here.

File: server/controllers/article.js (hunk 1 above)
Comment (TheoOkafor) [reply server/controllers/article.js:209]: I have made the required changes.

**Comment by seyi-adeleke** [comment]
The json response in the screenshot looks weird. Data should be an object

**Review by seyi-adeleke — APPROVED** [review]

**Pull request merged**